### PR TITLE
[main][test] add example when calling work as whitelisted contract

### PR DESCRIPTION
**Pull request by ricardo-alpaca**
## Description
Add an example on how to call `Vault.work()` from a whitelisted contract to open a position.

## Why?
Because current test has not cover this scenario and this would be a starting example for any protocol that wish to integrate with Alpaca's LYF product.

### CODE REVIEW

File: contracts/6/protocol/mock/MockWorkCaller.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
+pragma solidity 0.6.6;
+
+import "../Vault.sol";
```
Comment (spicysquid168) [inline contracts/6/protocol/mock/MockWorkCaller.sol:3]: Use `IVault` instead

File: contracts/6/protocol/mock/MockWorkCaller.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    uint256 principalAmount,
+    uint256 borrowAmount,
+    uint256 maxReturn,
+    address strat,
+    uint256 farmingTokenAmount,
+    uint256 minLpAmount
+  ) external returns (uint256 positionID) {
+    positionID = vault.nextPositionID();
+
+    vault.token().safeApprove(address(vault), uint256(-1));
```
Comment (spicysquid168) [inline contracts/6/protocol/mock/MockWorkCaller.sol:21]: move this to constructor

**Review by spicysquid168 — CHANGES_REQUESTED** [review]

File: contracts/6/protocol/mock/MockWorkCaller.sol (hunk 1 above)
Comment (ricardo-alpaca) [reply contracts/6/protocol/mock/MockWorkCaller.sol:3]: Well, `IVault` does not have `nextPositionID()`. So, I used `Vault` instead.

**Review by alpanaca — DISMISSED** [review]

File: contracts/6/protocol/mock/MockWorkCaller.sol (hunk 1 above)
Comment (ricardo-alpaca) [reply contracts/6/protocol/mock/MockWorkCaller.sol]: chane to IVault02

**Review by alpanaca — APPROVED** [review]

**Pull request closed without merging**